src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -753,11 +753,11 @@ bool ShenandoahHeap::is_in(const void* p) const {
 }
 
 bool ShenandoahHeap::is_in_young(const void* p) const {
-  return heap_region_containing(p)->affiliation() == ShenandoahRegionAffiliation::YOUNG_GENERATION;
+  return is_in(p) && heap_region_containing(p)->affiliation() == ShenandoahRegionAffiliation::YOUNG_GENERATION;
```

Comment:
Is there a valid case where we would check off-heap pointers? If not, then better do this as assert?